### ISSUE TITLE
We are NOT flushing as expected

### ISSUE THREAD

**Issue by rob-c** [ganga-devs/ganga#260]
We are NOT flushing to disk upon job state transitions, or as far as I can tell at all until shutdown.

I've tested this by submitting jobs with monitoring on/off and have hit ```Ctrl+Z``` with a live ganga session to examine the ```data``` file. Aside from the forced flush on the job creating we have flushed no knowledge of the job to disk.
(even after 30sec as far as I can tell)

**Comment by rob-c** [comment]
As an aside doing the following exposes some fragility which needs examining:

```
> ganga ##launch ganga interactively
...
> Ctrl+Z ##push ganga to a background processin bash
> fg ##bring it back to the foreground
```
This exposes lockfiles being destroyed by the version of ganga which is supposed to look after it. (or discovers a file never exited upon inspection).

**Comment by drmarkwslater** [comment]
The main issue here was fixed with PR #267 . The pushing to background problem probably needs a separate issue.
